### ISSUE TITLE
Why rethrow uncaught exceptions as UncaughtException

### ISSUE THREAD

**Issue by younishd** [chriso/klein.php#314]
Why would Klein catch any uncaught exception and rethrow it as `Klein\Exceptions\UnhandledException`?

I'm trying to do something like this (simplified):

```php
try {
    // ...
    $klein->dispatch();
} catch (SomeException $e) {
    // handle exception
}
```

The problem is that any Exception that is thrown inside `dispatch()` seems to be rethrown by Klein as `UnhandledException`.

So I was wondering why would a router class even do that kind of stuff in the first place (i.e., care about uncaught exceptions) and if there is a good reason maybe (I know PHP is kind of a mess with error/exception handling).

Now my workaround code looks like this:

```php
try {
    try {
        // ...
        $klein->dispatch();
    } catch (UnhandledException $e) {
        // rethrow
        throw new SomeException($e->getMessage());
    }
} catch (SomeException $e) {
    // handle exception
}
```

Is there a way to disable this behavior or maybe a (better) workaround for this problem, and if there is none, then why does Klein have to do this?

**Comment by jk3us** [comment]
This is related to #298, and I agree that I don't see a reason to catch uncaught exceptions at that level (that's what [`set_exception_handler`](https://secure.php.net/manual/en/function.set-exception-handler.php) is for).

**Comment by ispieter** [comment]
I agree with this matter. It's messing up my current exception handler. I don't think it's the task of the router to handle those Exceptions.

**Comment by whackashoe** [comment]
@Rican7 I'm hitting this too, I WANT to propagate exceptions all the way up. What about a way to disable this?

**Comment by alessandrob88** [comment]
I also have the same problem. I would like to detect exceptions and which file have generated them. How can we resolve this issue?
